src/components/Drawer/Drawer.component.tsx

```diff
@@ -1,18 +1,20 @@
 import React from 'react';
-import { DisclosureProps, Disclosure, DisclosureContent, useDisclosureState } from 'reakit';
-import { ButtonProps } from '../Button/Button';
+import { Disclosure, DisclosureContent, useDisclosureState } from 'reakit';
+import Button, { ButtonComponentType } from '../Button';
 
 import * as S from './Drawer.style';
+import { ButtonProps } from '../Button/Button';
 
-export type DrawerProps = DisclosureProps & {
-	disclosure: ButtonProps;
-	heading?: React.ComponentPropsWithRef<any>;
-	footer?: React.ComponentPropsWithRef<any>;
+export type DrawerProps = {
+	toggleButton?: React.ReactElement<ButtonProps, ButtonComponentType>;
```

Comment:
Two things here: 
- `disclosure` seems technical, before Reakit I wouldn't have known what it means. `toggleButton` seems clearer.
- The type is purely indicative (code as documentation), since Typescript cannot infer the type of an instantiated component  passed as a prop. You *can* give `toggleButton` anything, sadly. We could try and do some sort of runtime validation here maaaaybe, but it seems overkill to me.

Reply:
Up to you "disclosure" is used by WCAG actually